src/synapse.py

```diff
@@ -83,6 +91,47 @@ def execute_migrate_config(self, container: ops.Container) -> None:
                 "Migrate config failed, please review your charm configuration"
             )
 
+    def server_name_configured(self, container: ops.Container) -> str | None:
+        """Get server_name from configuration file.
+
+        Args:
+            container: Container of the charm.
+
+        Returns:
+            str | None: server_name or None if configuration file is not found.
+        """
+        try:
+            configuration_content = str(
+                container.pull(SYNAPSE_CONFIG_PATH, encoding="utf-8").read()
+            )
+        except PathError:
+            logger.error("configuration file %s does not exist", SYNAPSE_CONFIG_PATH)
+            return None
+        return yaml.safe_load(configuration_content)["server_name"]
+
+    def reset_instance_action(self, container: ops.Container) -> None:
+        """Erase data and config server_name.
+
+        Args:
+            container: Container of the charm.
+
+        Raises:
+            PathError: if somethings goes wrong while erasing the Synapse directory.
+        """
+        logging.debug("Erasing directory %s", SYNAPSE_CONFIG_DIR)
+        try:
+            container.remove_path(SYNAPSE_CONFIG_DIR, recursive=True)
+        except PathError as path_error:
+            # The error "unlinkat //data: device or resource busy" is expected
```

Comment:
Would it be worth elaborating on why it is expected?

Reply:
Sure, added a note about that.